pkg/sql/logictest/tests/fakedist/fakedist_test: TestLogic_zone_config_system_tenant failed

pkg/sql/logictest/tests/fakedist/fakedist_test.TestLogic_zone_config_system_tenant [failed](https://tanzanite.cluster.engflow.com/invocations/default/7a1da3ab-c1f9-4b22-ad67-762dc3bf41f5?testReportRun=1&testReportShard=32&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdDpmYWtlZGlzdF90ZXN0) on release-25.1.3-rc @ [70ba2ee1e23ea1c6a8b6c3983164857f5cc97960](https://github.com/cockroachdb/cockroach/commits/70ba2ee1e23ea1c6a8b6c3983164857f5cc97960):


```
=== RUN   TestLogic_zone_config_system_tenant
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLogic_zone_config_system_tenant906412735
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:139: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    logic.go:4440: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_zone_config_system_tenant906412735
--- FAIL: TestLogic_zone_config_system_tenant (73.90s)
=== RUN   TestLogic_zone_config_system_tenant/regression_93614
    logic.go:3127: let $t_id = 110
[11:19:13] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant: 22 statements
    logic.go:3065: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant:137: SELECT
          crdb_internal.pretty_key(start_key, -1),
          crdb_internal.pb_to_json('cockroach.roachpb.SpanConfig', config)
        FROM system.span_configurations
        WHERE end_key > (SELECT crdb_internal.table_span(110)[1])
        ORDER BY start_key
        expected:
            
        but found (query options: "retry") :
            /Table/110  {"gcPolicy": {"ttlSeconds": 90001}, "numReplicas": 3, "rangeMaxBytes": "67108864", "rangeMinBytes": "1048576"}
            /Table/111  {"gcPolicy": {"ttlSeconds": 90001}, "numReplicas": 3, "rangeMaxBytes": "1073741824", "rangeMinBytes": "67108864"}
            
[11:20:00] --- progress: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant: 27 statements
[11:20:00] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist/fakedist_test_/fakedist_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant with config fakedist: 27 tests, 1 failures
[11:20:07] --- total progress: 27 statements
--- total: 27 tests, 1 failures
    --- FAIL: TestLogic_zone_config_system_tenant/regression_93614 (50.80s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=32</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_zone_config_system_tenant.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48872

--- a/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant
+++ b/pkg/sql/logictest/testdata/logic_test/zone_config_system_tenant
@@ -141,6 +141,18 @@ FROM system.span_configurations
 WHERE end_key > (SELECT crdb_internal.table_span($t_id)[1])
 ORDER BY start_key
 ----
+/Table/110  {"gcPolicy": {"ttlSeconds": 90001}, "numReplicas": 3, "rangeMaxBytes": "67108864", "rangeMinBytes": "1048576"}
+/Table/111  {"gcPolicy": {"ttlSeconds": 90001}, "numReplicas": 3, "rangeMaxBytes": "1073741824", "rangeMinBytes": "67108864"}
+
+# Run the same query again and make sure there are 2 rows. This assertion is
+# only here to prevent the previous test case from being rewritten accidentally.
+statement count 2
+SELECT
+  crdb_internal.pretty_key(start_key, -1),
+  crdb_internal.pb_to_json('cockroach.roachpb.SpanConfig', config)
+FROM system.span_configurations
+WHERE end_key > (SELECT crdb_internal.table_span($t_id)[1])
+ORDER BY start_key
 
 subtest transactional_schemachanges
 

